VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseDiscussionResponsesAdapter.java

```diff
@@ -292,8 +293,21 @@ public void onException(Exception ex) {
         holder.socialLayoutViewHolder.threadFollowContainer.setVisibility(View.INVISIBLE);
 
         if (comment.isEndorsed()) {
+            DiscussionThread.ThreadType threadType = discussionThread.getType();
+            @StringRes int stringRes;
+            switch (threadType) {
+                case QUESTION:
+                    holder.responseAnswerTextView.setText(R.string.discussion_responses_answer);
```

Comment:
Instead of calling `holder.responseAnswerTextView.setText()` in both cases, a variable holding the string resource ID could be assigned for this as well (called something like `endorsementTypeStringRes`), and the `setText()` call performed once at the end.